x-pack/plugins/observability/server/routes/slo/route.ts

```diff
@@ -275,7 +275,15 @@ const getDiagnosisRoute = createObservabilityServerRoute({
     const esClient = (await context.core).elasticsearch.client.asCurrentUser;
     const licensing = await context.licensing;
 
-    return getGlobalDiagnosis(esClient, licensing);
+    try {
+      const response = await getGlobalDiagnosis(esClient, licensing);
+      return response;
+    } catch (error) {
+      if (error.cause.statusCode === 403) {
+        throw forbidden('Insufficient Elasticsearch cluster permissions to access feature.');
+      }
+      throw failedDependency(error);
```

Comment:
what does this thing do?

Reply:
It's one of HAPI Boom's methods to auto-format the response:

```
Boom.failedDependency([message], [data])
Returns a 424 Failed Dependency error where:

message - optional message.
data - optional additional error data.
Boom.failedDependency('an external resource failed');
Generates the following response payload:

{
    "statusCode": 424,
    "error": "Failed Dependency",
    "message": "an external resource failed"
}
```

https://hapi.dev/module/boom/api/?v=10.0.1#boomfaileddependencymessage-data